Apps/Src/CAN_Queue.c

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2020 UT Longhorn Racing Solar */
+/** CAN_Queue.c
+ * Queue that holds all CAN messages that Task_CANBusConsumer needs to send.
+ */
+
+#include "CAN_Queue.h"
+#include "os.h"
+#include "CANbus.h"
+#include "Tasks.h"
+
+// fifo
+#define FIFO_TYPE CANMSG_t
+#define FIFO_SIZE 256
+#define FIFO_NAME CAN_fifo
+#include "fifo.h"
+
+static CAN_fifo_t canFifo;
+static OS_SEM canFifoSem;
+static OS_MUTEX canFifoMutex;
+
+void CAN_Queue_Init(void) {
+    OS_ERR err;
+    CPU_TS ticks;
+    OSMutexCreate(&canFifoMutex, "CAN queue mutex", &err);
+    assertOSError(err);
+    OSSemCreate(&canFifoSem,
+                "CAN queue semaphore",
+                0,
+                &err);
+    assertOSError(err);
+    OSMutexPend(&canFifoMutex, 0, OS_OPT_POST_NONE, &ticks, &err);
+    assertOSError(err);
+    CAN_fifo_renew(&canFifo);
+    OSMutexPost(&canFifoMutex, OS_OPT_POST_NONE, &err);
+    assertOSError(err);
+}
+
+void CAN_Queue_Post(CANMSG_t message) {
+    OS_ERR err;
+    CPU_TS ticks;
+    OSMutexPend(&canFifoMutex, 0, OS_OPT_POST_NONE, &ticks, &err);
```

Comment:
Place comments here for what to do if CAN_fifo_put returns error.

Reply:
Changed this to return an `ErrorStatus` as discussed in-person